### ISSUE TITLE
Process with multiple Start Events

### ISSUE THREAD

**Issue by sashayar13** [sartography/spiff-arena#922]
**Scenario**
There is a process (Upload expense evidence) which can be started through a number of ways:
- Manually by a user
- Message received from Request Travel process
- Message received from Request Goods/Services process

These are the process prototypes I use:
https://dev.app.spiff.status.im/process-models/misc:prototypes:upload-expenses-start:upload-expense-evidense-v2

https://dev.app.spiff.status.im/process-models/misc:prototypes:upload-expenses-start:send-message-to-start-event-purchase

https://dev.app.spiff.status.im/process-models/misc:prototypes:upload-expenses-start:send-message-to-start-event-travel

The current configuration of the upload expense evidence process is not functioning correctly. Whenever the process starts manually or by one of the message events, another event is left in a "waiting" state. I am open to any suggestions on how to achieve the desired solution. Although the option with three separate processes and a call activity is available, I would like to explore other alternatives as well.

**Comment by MarD0607** [comment]
The common denominator is the upload expense process, and we want that this process can be started by an user and also started by a message event.
In this case, my suggestions would be the following:
- upload expense has a message start event
- the user manually started process would be a simple process that collects the user data and any other data needed and then as an end step sends a message throw event to the upload expense process. This for the user should be transparent and he will be carried to the next user task included in the upload expense 
- then of course the other processes that need to call the upload expense process will do it in the same way through a message throw event
So basically we need to define the upload expense process in such a way that it can be easily called by any other process and this should be transparent to the user.

**Comment by sashayar13** [comment]
> The common denominator is the upload expense process, and we want that this process can be started by an user and also started by a message event. In this case, my suggestions would be the following:
> 
> * upload expense has a message start event
> * the user manually started process would be a simple process that collects the user data and any other data needed and then as an end step sends a message throw event to the upload expense process. This for the user should be transparent and he will be carried to the next user task included in the upload expense
> * then of course the other processes that need to call the upload expense process will do it in the same way through a message throw event
>   So basically we need to define the upload expense process in such a way that it can be easily called by any other process and this should be transparent to the user.

@MarD0607 would imagine that with this approach, the Upload expense evidence process can have just one "Message Start Event"?

**Comment by MarD0607** [comment]
@sashayar13 Yes, so only in the case when the Upload expense evidence process steps are not different depending on where the start of the process originates.
If that assumption is true, then I assume the Upload expense evidence process can have one Message Start event that will be called from any process.

**Comment by essweine** [comment]
The library changes that need to be made to make this work are in https://github.com/sartography/SpiffWorkflow/pull/381

**Comment by sashayar13** [comment]
The fix is WIP, I will create 3 separate process models with Call Activities instead

**Comment by madhurrya** [comment]
@burnettk Is this already merged to dev.app?

**Comment by burnettk** [comment]
@madhurrya yeah

**Comment by dinithihj** [comment]
@sashayar13  I have tested this, below are the screenshots of the diagrams for each scenario. As expected other two start events are getting cancelled now

- Manually by a user
![image](https://github.com/sartography/spiff-arena/assets/123297840/902125a9-b6d5-49bd-9f9f-e8b9d2a64db6)

- Message received from Request Travel process
![image](https://github.com/sartography/spiff-arena/assets/123297840/01da3f9a-50cd-4733-bcbb-c996b75ec535)

- Message received from Request Goods/Services process
![image](https://github.com/sartography/spiff-arena/assets/123297840/f2c05c74-7567-450d-87c4-21c081c9a7a0)

Can you please let me know whether you want me to test anything else related to this ticket?

**Comment by dinithihj** [comment]
@sashayar13 When I delete everything after the Manual Task those process instances moved to the Completed status. Below are the screenshots from the sample I created. 

- Manually by a user
![image](https://github.com/sartography/spiff-arena/assets/123297840/2936798f-179f-47ab-bd78-25eeaaa327a6)

- Message received from Process Model 1
![image](https://github.com/sartography/spiff-arena/assets/123297840/2f7b8afa-1bf3-43de-8770-8164740c578b)


- Message received from Process Model 2
![image](https://github.com/sartography/spiff-arena/assets/123297840/6e39d7a4-9d61-4356-8903-9c57c298b79a)


This is the Messages tab, There are no "Ready" messages for those instances.
![image](https://github.com/sartography/spiff-arena/assets/123297840/2e7ddad3-6902-4336-8191-c038a0dbd8e1)

**Comment by sashayar13** [comment]
perfect, thanks for checking @dinithihj

**Comment by dinithihj** [comment]
Tested and verified on test.app (version 193/195).
Tested using below sample models

- https://test.app.spiff.status.im/process-models/misc:qa:multiple-start-event-test-model
- https://test.app.spiff.status.im/process-models/misc:qa:multiple-start-events-test-call-1
- https://test.app.spiff.status.im/process-models/misc:qa:multiple-start-events-test-call-2

**Comment by madhurrya** [comment]
Released to Prod on 2024 February 20th
